Presentation.Web/Infrastructure/APIRequestsLoggingMiddleware.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web.Security;
+using Microsoft.Owin;
+using Ninject;
+using OwinNinjectExample.Ninject;
+using Serilog;
+
+namespace Presentation.Web.Infrastructure
+{
+    public class ApiRequestsLoggingMiddleware : OwinMiddleware
+    {
+
+        private ILogger _logger = Log.Logger;
+
+        public ApiRequestsLoggingMiddleware(OwinMiddleware next) : base(next)
+        {
+        }
+
+        public override async Task Invoke(IOwinContext context)
+        {
+            _logger = context.GetResolutionScope().Get<ILogger>();
+            if (context.Request.Headers.ContainsKey("Authorization"))
+            {
+                var startTime = DateTime.Now;
+                var route = context.Request.Path;
+                var method = context.Request.Method;
+                var queryParameters = getQueryParameters(context.Request.Query);
+                var userID = context.Request.User.Identity.Name;
```

Comment:
Du kan se på brugeren om han er auithenticated med en token. Der ligger et claim af typen "TokenAuth" med navnet "<brugerens id>". Hvis DEN er på brugeren, SÅ er brugeren authenticated med token. Din ydre betingelse er dog fin, men forudsætningen for at brugeren er authenticated med token er den jeg nævner her :-)

Reply:
Kan umiddelbart ikke finde det claim. Kan dog finde AuthenticationType som er sat til JWT.

Reply:
Ahh ja men det er også nok.. det er den der "AddBearerTokenAuthentication" funktion der sætter auth type. Men det er fino og det er det vi skal se på som betingelse